### PR TITLE
Integrating Directus with Spring Boot

### LINKED ISSUE #52: Integrating Directus with Spring Boot
### What is your idea?

Spring Boot is one of the most used Java development frameworks. This tutorial will show the reader how to integrate Directus with Spring Boot. We will create an article API which will then be consumed by Spring Boot and displayed using the Thymeleaf templating engine.

### What are the key takeaways from your post?

- How to create a Spring Boot project.
- How to create a DTO for an article
- How to consume Directus API.
- How to display articles using Thymeleaf.

### What type of post is this?

Getting Started With Directus and {Language/Framework/Environment}

### Country of residence

Kenya

### Terms & Conditions

- [X] You understand that submitting this form does not mean we accept your proposal. We will accept proposals at our discretion.
- [X] You understand that to be paid you will need to generate an invoice to us once the final draft of your work is completed.
- [X] You understand that you will grant us an unlimited license to post and promote your work as part of this program. You may repost, but using our copy as the canonical source.
- [X] You are aged 18 or over.

### PR DESCRIPTION
**Pull request by davidmbochi**
closes #52

### CODE REVIEW

**Comment by davidmbochi** [comment]
Okay. Thanks for the update.

**Review by phazonoverload — CHANGES_REQUESTED** [review]
Left some changes for you to make :)

File: directus-spring-boot/integrating spring boot in directus.md [hunk 1]
```diff
@@ -0,0 +1,271 @@
+Spring Boot is one of the most used Java development frameworks and it comes in handy when developing web applications. This framework provides different libraries and tools that you can leverage to integrate the Directus API with your application.
+
+As a result, this tutorial will show you how to integrate Spring Boot in Directus using an article API. At the end of this tutorial, you will have an application that uses Thymeleaf templating engine to display the data from the API.
+
+## Prerequisites
+- Knowledge of Java programming language.
+- Basic knowledge of Thymeleaf
+- [Java development kit(JDK)8](https://www.oracle.com/ke/java/technologies/javase/javase8-archive-downloads.html) and above.
+- Directus project - Use the [quickstart guide](https://docs.directus.io/getting-started/quickstart) to create a project with the fields `id`, `title`, `featuredImage`, and `body`.
+- [IntelliJ IDEA community edition](https://www.jetbrains.com/idea/download/#section=linux) - Development environment for the Java programming language.
```
[the remark this answers is not in the document]
Comment (davidmbochi) [reply directus-spring-boot/integrating spring boot in directus.md:7]: The IDE is usually provided for people not familiar with Java and don't know the IDE to use. However, we can remove it if the target is for Junior and senior developers.

File: directus-spring-boot/integrating spring boot in directus.md [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+
+## Prerequisites
+- Knowledge of Java programming language.
+- Basic knowledge of Thymeleaf
+- [Java development kit(JDK)8](https://www.oracle.com/ke/java/technologies/javase/javase8-archive-downloads.html) and above.
+- Directus project - Use the [quickstart guide](https://docs.directus.io/getting-started/quickstart) to create a project with the fields `id`, `title`, `featuredImage`, and `body`.
+- [IntelliJ IDEA community edition](https://www.jetbrains.com/idea/download/#section=linux) - Development environment for the Java programming language. 
+
+
+## Project setup
```
[the remark this answers is not in the document]
Comment (davidmbochi) [reply directus-spring-boot/integrating spring boot in directus.md:10]: Noted.

File: directus-spring-boot/integrating spring boot in directus.md [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+
+To add the project dependencies, click on the **ADD DEPENDENCIES** button then add the dependencies described below. In the search bar that opens, search for the name of the dependency then select it to include it in the project. Repeat the same process for all the dependencies.
+
+- Spring Web - Spring Web dependency provides the functionalities required to build a web application using the MVC pattern. This dependency also provides support for RESTful architecture and uses Apache Tomcat as the default server.
+
+- Lombok - Lombok helps to reduce boilerplate code by generating getter and setter methods for your application.
+
+- Thymeleaf - Thymeleaf is a templating engine that will help you to display dynamic data from the API on the HTML pages.
+
+If you followed this steps to the latter, the project structure and dependencies of your application should be as shown below.
```
[the remark this answers is not in the document]
Comment (davidmbochi) [reply directus-spring-boot/integrating spring boot in directus.md:25]: Sure

File: directus-spring-boot/integrating spring boot in directus.md [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+
+![project structure](create-spring-boot-project.png)
+
+Next, click the **GENERATE** button to generate a zip file of your project. Extract the zip file to a desired location on your computer then import the extracted Spring Boot project in IntelliJ.
+
+## Create project sub packages
+Having different packages in your application will help you to separate responsibilities, improve the readability of the code, and simplify making changes.
```
[the remark this answers is not in the document]
Comment (davidmbochi) [reply directus-spring-boot/integrating spring boot in directus.md:32]: Noted

File: directus-spring-boot/integrating spring boot in directus.md [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+    private Long id;
+
+    private String title;
+
+    private String featuredImage;
+
+    private String body;
+}
+```
+In this file, you have defined a class named `Article` with the fields `id`, `title`, `featuredImage`, and `body`. Note that this fields should be similar to those in your Directus project.
```
[the remark this answers is not in the document]
Comment (davidmbochi) [reply directus-spring-boot/integrating spring boot in directus.md:55]: Okay.

File: directus-spring-boot/integrating spring boot in directus.md [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+
+As a result, this tutorial will show you how to integrate Spring Boot in Directus using an article API. At the end of this tutorial, you will have an application that uses Thymeleaf templating engine to display the data from the API.
+
+## Prerequisites
+- Knowledge of Java programming language.
+- Basic knowledge of Thymeleaf
+- [Java development kit(JDK)8](https://www.oracle.com/ke/java/technologies/javase/javase8-archive-downloads.html) and above.
+- Directus project - Use the [quickstart guide](https://docs.directus.io/getting-started/quickstart) to create a project with the fields `id`, `title`, `featuredImage`, and `body`.
+- [IntelliJ IDEA community edition](https://www.jetbrains.com/idea/download/#section=linux) - Development environment for the Java programming language. 
+
```
[the remark this answers is not in the document]
Comment (davidmbochi) [reply directus-spring-boot/integrating spring boot in directus.md:8]: Okay.

File: directus-spring-boot/integrating spring boot in directus.md [hunk 7]
```diff
@@ -0,0 +1,271 @@
...
+"data": [
+{
+"id": 2,
+"title": "Integrating Directus in Spring Boot",
+"featuredImage": "f1b20633-5329-4a6a-8942-1983a5732e7e",
+"body": "In this tutorial, you will learn how to integrate Directus in Spring Boot by leveraging Thymeleaf templating engine"
+}
+]
+}
+```
```
[the remark this answers is not in the document]
Comment (davidmbochi) [reply directus-spring-boot/integrating spring boot in directus.md:73]: Noted

**Comment by davidmbochi** [comment]
Hi Kevin.

I have applied the requested changes. You can now review and let me know.

File: directus-spring-boot/integrating spring boot in directus.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (phazonoverload) [reply directus-spring-boot/integrating spring boot in directus.md]: And yes you're right if this was a beginner Java post, but we've already said they should have knowledge so it's safe to assume they have an IDE set up.

**Review by phazonoverload — APPROVED** [review]

**Comment by phazonoverload** [comment]
Thank you for writing this, and I'm happy to consider it done in it's current state. I'll schedule it in to be published. 

This repo is just for authoring/reviewing - the actual piece will be published via a Directus project, so I'll take it from here. I'll close this issue to keep the main branch clean, but this is a acceptance of your work.

Could you email me at [kevin@directus.io](mailto:kevin@directus.io) and I'll share what we need to get you paid.

**Pull request closed without merging**